healthcheck/utils.py

```diff
@@ -2,25 +2,21 @@
 import hashlib
 import os
 
-from django.conf import settings
 from google.cloud import bigquery
 from google.oauth2 import service_account
 from iso6709 import Location
 
 
-bigquery_client = None
-if os.path.isfile(settings.TBCONNECT_BQ_KEY_PATH):
-    credentials = service_account.Credentials.from_service_account_file(
-        settings.TBCONNECT_BQ_KEY_PATH,
-        scopes=["https://www.googleapis.com/auth/cloud-platform"],
-    )
+def get_bigquery_client(key_path):
+    if os.path.isfile(key_path):
+        credentials = service_account.Credentials.from_service_account_file(
+            key_path, scopes=["https://www.googleapis.com/auth/cloud-platform"],
+        )
 
-    bigquery_client = bigquery.Client(
-        credentials=credentials, project=credentials.project_id,
-    )
+        return bigquery.Client(credentials=credentials, project=credentials.project_id,)
```

Comment:
This means that we'll be creating a new client every time this function is called, which might be an issue if the client reuses connections or something. I think if we just put the `lru_cache` decorator on this function, it should cache the client per credentials file.

Reply:
Thanks @rudigiesler ready for another look